src/modules/elixir.rs

```diff
@@ -0,0 +1,81 @@
+use ansi_term::Color;
+use lazy_static::lazy_static;
+use regex::Regex;
+use std::process::Command;
+
+use super::{Context, Module};
+
+const ELIXIR_VERSION_PATTERN: &str = "\
+Erlang/OTP (?P<otp>\\d+)[^\\n]+
+
+Elixir (?P<elixir>\\d[.\\d]+).*";
+
+lazy_static! {
+    static ref ELIXIR_VERSION_REGEX: Regex = Regex::new(ELIXIR_VERSION_PATTERN).unwrap();
+}
+
+/// Create a module with the current Elixir version
+///
+/// Will display the Rust version if any of the following criteria are met:
+///     - Current directory contains a `mix.exs` file
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let is_elixir_project = context.try_begin_scan()?.set_files(&["mix.exs"]).is_match();
+
+    if !is_elixir_project {
+        return None;
+    }
+
+    let (otp_version, elixir_version) = get_elixir_version()?;
+
+    let mut module = context.new_module("elixir");
+    let module_style = module
+        .config_value_style("style")
+        .unwrap_or_else(|| Color::Purple.bold());
+    module.set_style(module_style);
+
+    module.new_segment("symbol", "🔮 ");
+    module.new_segment(
+        "version",
+        &format!("{} (OTP {})", elixir_version, otp_version),
+    );
+
+    Some(module)
+}
+
+fn get_elixir_version() -> Option<(String, String)> {
+    let output = Command::new("elixir")
+        .arg("--version")
+        .output()
+        .ok()
+        .and_then(|output| String::from_utf8(output.stdout).ok())?;
+
+    parse_elixir_version(&output)
+}
+
+fn parse_elixir_version(version: &str) -> Option<(String, String)> {
```

Comment:
As someone who has no idea how elixir works: what's the OTP version?

Reply:
OTP it the name of the Erlang runtime system and library (Erlang being the "core" of Elixir, which it compiles to), you can have more info on the OTP repository: https://github.com/erlang/otp

> OTP is a set of Erlang libraries, which consists of the Erlang runtime system, a number of ready-to-use components mainly written in Erlang, and a set of design principles for Erlang programs. [Learn more about Erlang and OTP.](http://erlang.org/doc/system_architecture_intro/sys_arch_intro.html)

Reply:
> the OTP version Elixir was compiled with and not the one actually instead

My reasoning for that was that the OTP version used to compile Elixir impacts an Elixir project. One example I've seen is that people with Erlang/OTP 21 installed [encountered problems](
https://github.com/elixir-lang/elixir/issues/5851#issuecomment-406853833) running Dialyzer on Elixir 1.6.6-otp-19.

In that case the output of `elixir -v` was:
```
Erlang/OTP 21 [erts-10.0.1] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:1]
Elixir 1.6.6 (compiled with OTP 19)
```
and in order to get the new functionality, they needed to Elixir to be compiled on OTP 20+. This kind of issue is supposed to be rare, but it does happen.